Source/JavaScriptCore/heap/Heap.cpp

```diff
@@ -3328,4 +3328,61 @@ DEFINE_DYNAMIC_ISO_SUBSPACE_MEMBER_SLOW(moduleProgramExecutableSpace)
 
 } // namespace GCClient
 
+#if ASSERT_ENABLED
+static Lock strongMapLock;
+static HashMap<uintptr_t, String>* debugStrongMap;
+static std::atomic<int> strongLoggingDisabled;
+static std::atomic<uintptr_t> lastStrongID;
+
+StrongLoggingDisabledScope::StrongLoggingDisabledScope()
+{
+    if (!Options::logLiveStrongs())
+        return;
+    ++strongLoggingDisabled;
+}
+
+StrongLoggingDisabledScope::~StrongLoggingDisabledScope()
+{
+    if (!Options::logLiveStrongs())
+        return;
+    --strongLoggingDisabled;
+}
+
+uintptr_t debugReportLiveStrong()
+{
+    if (!Options::logLiveStrongs() || strongLoggingDisabled.load() > 0)
+        return 0;
+    Locker locker(strongMapLock);
+    if (!debugStrongMap)
+        debugStrongMap = new HashMap<uintptr_t, String>();
```

Comment:
This could be useful in WebCore, especially since we want to get rid of all uses of Strong there. Could this hash map instead live in `WTF` so we can get access to it from WebCore too?

I use this object on my branch: https://github.com/rreno/WebKit/blob/eng/ref-tracking/Source/WTF/wtf/RefTracker.h

and I give access to the hash map globally via the static getters.